crates/miden-objects/src/block/proven_block.rs

```diff
@@ -1,38 +1,26 @@
-use alloc::vec::Vec;
-
 use crate::MIN_PROOF_SECURITY_LEVEL;
-use crate::block::{BlockAccountUpdate, BlockBody, BlockHeader, OutputNoteBatch};
-use crate::note::Nullifier;
+use crate::block::{BlockBody, BlockHeader, BlockProof};
 use crate::utils::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable};
 
 // PROVEN BLOCK
 // ================================================================================================
 
-/// A block in the Miden chain.
+/// Represents a block in the Miden blockchain that has been signed and proven.
 ///
-/// A block is built from batches of transactions, i.e. multiple
-/// [`ProvenBatch`](crate::batch::ProvenBatch)es, and each batch contains multiple
-/// [`ProvenTransaction`](crate::transaction::ProvenTransaction)s.
+/// Blocks transition from proposed, signed, and proven states. This struct represents the final,
```

Comment:
```suggestion
/// Blocks transition through proposed, signed, and proven states. This struct represents the final,
```

Nit: The current sentence reads incomplete.